ide/lsp.client/src/org/netbeans/modules/lsp/client/LSPBindings.java

```diff
@@ -568,6 +599,10 @@ public void run() {
             if(! process.isAlive()) {
                 return;
             }
+            LSPBindings bindings = get();
+            if (bindings != null) {
+                bindings.worker.shutdownNow();
+            }
```

Comment:
I would be surprised if this works. `LSPReference` is a `WeakReference`, so when this is called, the object is already collected and not accessible from java anymore (at least I assume that his makes cleaning via `ReferenceQueue` easier to reason about than `#finalize`). I might be wrong though.